File: app/src/main/java/com/example/mvctutorial/designpattern/_11decorator/java/concrete/Milk.java
package com.example.mvctutorial.designpattern._11decorator.java.concrete;

import com.example.mvctutorial.designpattern._11decorator.java.abst.AbstAdding;
import com.example.mvctutorial.designpattern._11decorator.java.abst.IBeverage;

public class Milk extends AbstAdding {

    public Milk(IBeverage base) {
        super(base);
    }

    @Override
    public int getTotalPrice() {
        return super.getTotalPrice()+50;
    }
}

File: app/src/main/java/com/example/mvctutorial/designpattern/_11decorator/concrete/Milk.kt
package com.example.mvctutorial.designpattern._11decorator.concrete

import com.example.mvctutorial.designpattern._11decorator.abst.AbstAdding
import com.example.mvctutorial.designpattern._11decorator.abst.IBeverage

class Milk(val base: IBeverage): AbstAdding(base) {

    override fun getTotalPrice() = super.getTotalPrice() + 50

}